packages/terra-section-header/src/SectionHeader.jsx

```diff
@@ -0,0 +1,139 @@
+
+const cx = classNames.bind(styles);
+
+const KEYCODES = {
+  ENTER: 13,
+  SPACE: 32,
+};
+
+const propTypes = {
+  /**
+   * Text to be displayed on the SectionHeader.
+   */
+  title: PropTypes.string.isRequired,
+  /**
+   * Callback function triggered when the accordion icon is clicked.
+   */
+  onClick: PropTypes.func,
+  /**
+   * Whether the accordion icon should be displayed in its 'open' or 'closed' position.
+   */
+  isOpen: PropTypes.bool,
+  /**
+   * Optionally sets the heading level. One of `1`, `2`, `3`, `4`, `5`, `6`. Default `level=2`.
+   */
+  level: PropTypes.oneOf([1, 2, 3, 4, 5, 6]),
+};
+
+const defaultProps = {
+  onClick: undefined,
+  isOpen: false,
+  level: 2,
+};
+
+const isRecognizedKeyPress = event => ((event.nativeEvent.keyCode === KEYCODES.ENTER) || (event.nativeEvent.keyCode === KEYCODES.SPACE));
+
+class SectionHeader extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = { isActive: false };
+    this.wrapOnKeyDown = this.wrapOnKeyDown.bind(this);
+    this.wrapOnKeyUp = this.wrapOnKeyUp.bind(this);
+  }
+
+  wrapOnKeyDown(onKeyDown) {
+    return ((event) => {
+      if (isRecognizedKeyPress(event) && !this.state.isActive) {
+        this.setState({ isActive: true });
+
+        // Call the same function the user provides for a click event
+        if (this.props.onClick) {
+          this.props.onClick(event);
+        }
+      }
+
+      if (onKeyDown) {
+        onKeyDown(event);
+      }
+    });
+  }
+
+  wrapOnKeyUp(onKeyUp) {
+    return ((event) => {
+      if (isRecognizedKeyPress(event) && this.state.isActive) {
+        this.setState({ isActive: false });
+      }
+
+      if (onKeyUp) {
+        onKeyUp(event);
+      }
+    });
+  }
+
+  render() {
+    const {
+      title,
+      onClick,
+      isOpen,
+      level,
+      ...customProps
+    } = this.props;
+
+    if ((process.env.NODE_ENV !== 'production') && (!onClick && isOpen)) {
+      // eslint-disable-next-line no-console
+      console.warn('\'isOpen\' are intended to be used only when \'onClick\' is provided.');
+    }
+
+    const attributes = Object.assign({}, customProps);
+
+    if (onClick) {
+      attributes.tabIndex = '0';
+      attributes.onKeyDown = this.wrapOnKeyDown(attributes.onKeyDown);
+      attributes.onKeyUp = this.wrapOnKeyUp(attributes.onKeyUp);
+    }
+
+    const iconClassNames = cx([
+      'accordion-icon',
+      { 'is-open': onClick && isOpen },
+    ]);
+
+    const accordionIcon = (
+      <div className={cx('accordion-icon-wrapper')}>
+        <span className={iconClassNames} />
+      </div>
+    );
+
+    const sectionHeaderClassNames = cx([
+      'section-header',
+      { 'is-interactable': onClick },
+      { 'is-active': this.state.isActive },
+      customProps.className,
+    ]);
+
+    const Element = `h${level}`;
+
+    // allows us to set an onClick on the div
+    /* eslint-disable jsx-a11y/no-static-element-interactions */
```

Comment:
Will none of these roles disable the label? It almost feels like we could potentially use button or switch, since it opens and closes the div? 

https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/no-static-element-interactions.md#case-this-element-acts-like-a-button-link-menuitem-etc

Reply:
@bjankord and I talked. We think role="button" is appropriate here for when onClick is provided.

Reply:
Talked in person. We can add this later.